### ISSUE TITLE
Test arg `--pinentry-mode loopback` always used for gpg2.1

### ISSUE THREAD

**Issue by absoludity** [sobolevn/git-secret#503]
What are the steps to reproduce this issue?
-------------------------------------------
1. Setup gpg agent forwarding as per https://wiki.gnupg.org/AgentForwarding, with gpg2 on both machines
2. on remote machine where git-secret is installed, `export SECRETS_GPG_COMMAND=gpg2`
3. On the remote machine, execute `git secret reveal` in a repo with secrets configured

What happens?
-------------
```
$ git secret reveal
gpg: setting pinentry mode 'loopback' failed: Forbidden
gpg: decryption failed: No secret key
git-secret: abort: problem decrypting file with gpg: exit code 2: ...
```

What were you expecting to happen?
----------------------------------
Prompted for passphrase (via pinentry back on the client) and files decrypted.

Any other comments?
-------------------
If I revert https://github.com/sobolevn/git-secret/commit/b78afdeff92914e0ee32bb1626e062862e8e2e2b#diff-0e013d86c1d0742c9ea18dfddb80e094 (ie. so the arg `--pinentry-mode loopback` is not included) it works as expected, with pinentry on the client prompting for the passphrase and gpg2 decrypts with the key from the agent.

From #107, it appears this arg was added for this reason:

- "Fixed the fact that test did not run on modern gnupg versions (Exported the keys prior to running tests, enforced lookback pinentry on tests in order to be able to pipe passphrase)"

I'm not 100%, but it looks like it would be better to default to this arg if gpg 2.1 is being used *and* a test specific env var is set, or if gpg2.1 is being used *and* a user arg explicitly requested loopback? If this is the case, happy to contribute the fix. 

What versions of software are you using?
----------------------------------------
**Operating system:** (`uname -a`) Linux dockerdev 4.4.0-148-generic #174-Ubuntu SMP Tue May 7 12:20:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**`git-secret` path:** (`which git-secret`) `$HOME/bin/git-secret`

**`git-secret` version:** (`git secret --version`) 0.2.6

**`git` version:** (`git --version`) `git version 2.21.0`

**Shell type and version:** (`$SHELL --version`) 
```
GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**`gpg` version:** (`gpg --version`) …
```
$ echo $SECRETS_GPG_COMMAND
gpg2
$ gpg2 --version
gpg (GnuPG) 2.1.11
libgcrypt 1.6.5
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: ~/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2

```

**Comment by joshrabinowitz** [comment]
Hello @absoludity , i think this is related to the class of errors where git-secret uses the wrong --pinentry-mode setting for a use case.

I have worked up PR #490 specifically for this type of issue, can you take a look and see if using that PR and setting `$SECRETS_PINENTRY` can resolve your issue? From that PR:

    use SECRETS_PINENTRY to specify 'default', 'ask', 'cancel', 'error' or 'loopback'.

**Comment by joshrabinowitz** [comment]
@absoludity I think this is resolved since #490 was merged. Let me know if you still have issues.

**Comment by absoludity** [comment]
Looks perfect, I'll test later but assume this will indeed fix my issue. Thanks!
